vsphere/image/data/user-data.pkrtpl.hcl

```diff
@@ -0,0 +1,198 @@
+#cloud-config
```

Comment:
nit: can we keep the naming consistent with other packer files used in the project: `<name>.pkr.hcl` ? 
Ref: https://developer.hashicorp.com/packer/docs/templates/hcl_templates#code-organization)

This file could be just `user-data.pkr.hcl`

Reply:
Will do, thanks for the review.